syntax = "proto2";


enum TaskType {
    CLASSIFICATION = 0;
    OBJECT_DETECTION = 1;
    SEMANTIC_SEGMENTATION = 2;
}

/*use this for pipe.add_data*/
message dataset_meta {
    required TaskType task_type = 1;
    required string input_forrmat = 2;
    required int32 train_set_size = 3;
    required int32 test_set_size = 4;
    optional int32 num_classes = 5;
}

/*use this for pipe.add_data*/
message dataset_version_meta {
    required HashEntry train_set_hash = 1;
    required HashEntry test_set_hash = 2;
    required string train_set_version = 3;
    required string test_set_version = 4;
}

/*use this for pipe.lock_data*/
message tfrecord_meta {
    required int32 num_shards = 1;
    repeated HashEntry train_set = 2;
    repeated HashEntry test_set = 3;
    required string train_set_hash = 4;
    required string test_set_hash = 5;
}

/*use this wherever you hash a file to store filename: hash value pair*/
message HashEntry {
    required string filename = 1;
    required string hash = 2;
}